### ISSUE TITLE
PnP Starter Intranet v2.0 - Custom MasterPage?

### ISSUE THREAD

**Issue by mwiedemeyer** [SharePoint/PnP#1699]
Hi @FranckyC 
this looks like a nice sample, but I'm a little bit confused.
It seems Microsoft always try to convince the community to **not** customize the master page.
This solution contains a custom master page and is considered a best practice which somehow seems to be inconsistent.
Can you bring some light into this topic?

Thanks,
marco

**Comment by FranckyC** [comment]
Hi @mwiedemeyer here is the extract from the official documentation about this point:

Why use a customized master page? 

For SharePoint 2013 and 2016 (to a lesser extent), using a customized master page remains an acceptable practice since it’s still a part of the development model adapted for these versions. However, for SharePoint Online, this can be a controversial choice since Microsoft now (August 2017) deems that modifying the master page is a bad practice, since it could lead to incompatibilities with future version updates (since the updates are managed by Microsoft, not you). However, here are some of the reasons that made us use a customized master page, even with SharePoint Online:

1.	A drastic change to the development model

First off, in SharePoint Online, there is a tremendous difference between the old and new development approaches. Microsoft went from a dated ASP model based on physical servers to a completely client-based model based on JavaScript with the arrival of the SharePoint Framework. Consequently, we strongly believe that Microsoft will no longer significantly invest in improving the classic SharePoint master page (except for a few minor integrations). Should this occur, updating the master page should not pose a problem.

2.	Understanding the purpose of the master page

A master page is used first and foremost to provide a common HTML structure to all the pages. Even though we tried not to rely on it (by injecting JavaScript code for example), a customized master page remains an essential tool when it comes to creating a professional looking mobile design without having to face the needless and inevitable technical puzzles and other architectural compromises. To sum it up, using a master page does not compromise in any way the rest of the solution if we take advantage of it and play by the rules.
Furthermore, since the SharePoint Framework is still in infancy (as of August 2017) and does not provide a viable web content creation solution for a complete intranet (let’s not even mention managing multiple languages), we also did not have any other option to create a complete, mobile responsive intranet solution.

3.	Understanding corporate mindset towards intranet solutions

When it comes to intranet solutions, there are usually two philosophies regarding the development approach: certain companies would rather manage Microsoft’s visual updates themselves and would prefer that any SharePoint reference be hidden in order to maintain the integrity of their corporate identity, while others don’t. For the former, this is where using a master page makes complete sense. For the latter, it is preferable to stick to SharePoint’s default behaviour. However, doing so could make the intranet’s design look awkward (not pretty…)…

4.	Compatibility between all SharePoint versions

Using a customized master page lets you also obtain a single file that is compatible with SharePoint 2013, 2016 and SharePoint Online, which is not negligible.

Hope this help ;)
